### PR TITLE
【prim】Matmul double grad composite api

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types
Others 

### PR changes
 Others 
### Describe
matmul 二阶组合算子实现

### CODE REVIEW

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 1]
```diff
@@ -148,6 +149,12 @@ void sum_grad(const Tensor& x,
         }
       } else {
         axis_ = axis.GetData();
+        for (int64_t i = 0; i < axis_size; i++) {
+          if (axis[i] < 0) {
+            axis_[i] = axis[i] + x_dim_size;
+            std::cout << "axis_[" << i << "] = " << axis[i];
```
Comment (Charles-hit) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: 删掉打印相关的代码

File: python/paddle/fluid/tests/unittests/prim/prim/vjp/eager/test_comp_eager_matmul_double_grad.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+            dx_, dy_, ddout_ = desired(
+                self.primal0, self.primal1, self.trans_0, self.trans_1, d_type
+            )
+
+            np.testing.assert_allclose(
+                actual=dx,
+                desired=dx_,
+                rtol=TOLERANCE[d_type]['rtol'],
+                atol=TOLERANCE[d_type]['atol'],
+            )
```
Comment (Charles-hit) [inline python/paddle/fluid/tests/unittests/prim/prim/vjp/eager/test_comp_eager_matmul_double_grad.py:332]: 请添加三阶测试保证拆解后反向的正确性

File: paddle/fluid/prim/api/manual_prim/utils/utils.h [hunk 3]
```diff
@@ -78,11 +81,38 @@ static phi::DDim get_reduce_dims(const phi::DDim& x_dims,
   return get_reduce_dims_from_out(out_dims, x_dims);
 }
 
+static std::vector<int> get_reduce_dims(const Tensor& dx,
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/manual_prim/utils/utils.h:84]: why this?

File: python/paddle/fluid/tests/unittests/prim/prim/vjp/static/test_comp_matmul_double_grad.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    "float16": {"rtol": 1e-3, "atol": 1e-3},
+    "float32": {"rtol": 1e-6, "atol": 1e-6},
+    "float64": {"rtol": 1e-15, "atol": 1e-15},
+}
+
+
+@param.parameterized_class(
+    ('primal0', 'primal1', 'primal2', 'trans_0', 'trans_1', 'dtype'),
+    [
+        # (
```
Comment (JiabinYang) [inline python/paddle/fluid/tests/unittests/prim/prim/vjp/static/test_comp_matmul_double_grad.py:42]: Add to do here to uncomment it when we fix static mode

File: paddle/fluid/prim/utils/static/static_global_utils.h [hunk 5]
```diff
@@ -93,7 +93,7 @@ class StaticCompositeContext {
   StaticCompositeContext()
       : current_block_desc_(nullptr),
         generator_(new UniqueNameGenerator()),
-        skip_comp_ops_({"matmul_v2"}) {}
+        skip_comp_ops_({"matmul_v2", "matmul_v2_grad"}) {}
```
Comment (JiabinYang) [inline paddle/fluid/prim/utils/static/static_global_utils.h]: leave todo here to remove this when we fix static

**Review by JiabinYang — DISMISSED** [review]
some coments

File: paddle/fluid/prim/api/manual_prim/utils/utils.h (hunk 3 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/manual_prim/utils/utils.h:84]: when matmul used broadcast [1, 4, 3] * [2, 3, 4] =[2, 4, 4] ,x will be broadcast to [2, 4, 3] we need to compute the true dims [1, 4, 3] reduce dims will be [0]

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: done

Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: done

File: python/paddle/fluid/tests/unittests/prim/prim/vjp/eager/test_comp_eager_matmul_double_grad.py (hunk 2 above)
Comment (xiaoguoguo626807) [reply python/paddle/fluid/tests/unittests/prim/prim/vjp/eager/test_comp_eager_matmul_double_grad.py:332]: done

File: python/paddle/fluid/tests/unittests/prim/prim/vjp/static/test_comp_matmul_double_grad.py (hunk 4 above)
Comment (xiaoguoguo626807) [reply python/paddle/fluid/tests/unittests/prim/prim/vjp/static/test_comp_matmul_double_grad.py:42]: done

File: paddle/fluid/prim/utils/static/static_global_utils.h (hunk 5 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/utils/static/static_global_utils.h]: done

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by zhwesky2010 — DISMISSED** [review]
LGTM for run_unittests.sh

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by zhwesky2010 — APPROVED** [review]

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**